flink-metrics/flink-metrics-prometheus/src/main/java/org/apache/flink/metrics/prometheus/PrometheusPushGatewayReporter.java

```diff
@@ -67,25 +73,44 @@ public void open(MetricConfig config) {
 		}
 
 		pushGateway = new PushGateway(host + ':' + port);
-		log.info("Configured PrometheusPushGatewayReporter with {host:{}, port:{}, jobName: {}, randomJobNameSuffix:{}, deleteOnShutdown:{}}", host, port, jobName, randomSuffix, deleteOnShutdown);
+		log.info("Configured PrometheusPushGatewayReporter with {host:{}, port:{}, jobName:{}, randomJobNameSuffix:{}, deleteOnShutdown:{}, groupingKey:{}}",
+			host, port, jobName, randomSuffix, deleteOnShutdown, groupingKey);
+	}
+
+	@VisibleForTesting
+	static Map<String, String> parserGroupingKey(final String groupingKeyConfig) {
+		if (!groupingKeyConfig.isEmpty()) {
+			Map<String, String> groupingKey = new HashMap<>();
+			String[] kvs = groupingKeyConfig.split(",");
+			for (String kv : kvs) {
+				int idx = kv.indexOf("=");
+				if (idx < 0) {
+					continue;
+				}
+				groupingKey.put(kv.substring(0, idx), kv.substring(idx + 1));
+			}
+
+			return groupingKey;
+		}
+		return null;
 	}
 
 	@Override
 	public void report() {
 		try {
-			pushGateway.push(CollectorRegistry.defaultRegistry, jobName);
+			pushGateway.push(CollectorRegistry.defaultRegistry, jobName, groupingKey);
```

Comment:
is there a difference in behavior between passing null and an empty collection?

Reply:
I did some test and see the code of pushGateway SDK, both null and an empty collection will be ignored.

Reply:
In that case `parseGroupingKey` should not return null but an empty map instead.